packages/sidecar/src/cli.ts

```diff
@@ -0,0 +1,233 @@
+          // TODO: Check docker-compose.yml existence
+          // TODO: Check docker-compose.yaml existence
+          // TODO: Check `docker compose --version` existence
+          // TODO: Check `docker-compose --version` existence
+        } catch {
+          // pass
+        }
+      }
+      if (cmdArgs.length === 0) {
+        logger.error("Error: No command specified to run and could not infer the command automatically.");
+        process.exit(1);
+      }
+      const cmdStr = cmdArgs.join(" ");
+      logger.info(`Starting command: ${cmdStr}`);
+      // TODO: Run this _after_ the server is started
+      const runCmd = spawn(cmdArgs[0], cmdArgs.slice(1), {
+        cwd: process.cwd(),
+        shell,
+        windowsVerbatimArguments: true,
+        windowsHide: true,
+        // TODO: Consider using pipe and forwarding output as logs to Spotlight
+        stdio: "ignore",
+        env: {
+          ...process.env,
+          SENTRY_SPOTLIGHT: getSpotlightURL(port),
+          // This is not supported in all SDKs but worth adding
+          // for the ones that support it
+          SENTRY_TRACES_SAMPLE_RATE: "1",
+          //  Consider setting SENTRY_DSN to our hack server too?
+        },
+      });
+      runCmd.on("error", err => {
+        logger.error(`Failed to run ${cmdStr}:`);
+        logger.error(err);
+        process.exit(1);
+      });
+      runCmd.on("close", code => {
+        if (!code) {
+          logger.error(`${cmdStr} exited, terminating.`);
+        } else {
+          logger.error(`${cmdStr} exited with code ${code}.`);
+        }
+        process.exit(code);
+      });
+      const killRunCmd = () => {
+        runCmd.removeAllListeners();
+        runCmd.kill();
+      };
+      runCmd.on("spawn", () => {
+        runCmd.removeAllListeners("error");
+        process.on("SIGINT", killRunCmd);
+        process.on("SIGTERM", killRunCmd);
+        process.on("beforeExit", killRunCmd);
+      });
+
+      cmd = "tail";
+      cmdArgs = [];
+      // biome-ignore lint/suspicious/noFallthroughSwitchClause:
+      // Intentional fallthrough to start the event streaming after running the command
+    }
+    case "tail": {
+      const eventTypes = cmdArgs.length > 0 ? cmdArgs.map(arg => arg.toLowerCase()) : ["everything"];
+      for (const eventType of eventTypes) {
+        if (!SUPPORTED_TAIL_ARGS.has(eventType)) {
+          logger.error(`Error: Unsupported argument "${eventType}".`);
+          logger.error(`Supported arguments are: ${[...SUPPORTED_TAIL_ARGS].join(", ")}`);
+          process.exit(1);
+        }
+      }
+
+      const types = eventTypes.some(type => EVERYTHING_MAGIC_WORDS.has(type))
+        ? SUPPORTED_ENVELOPE_TYPES
+        : new Set([...eventTypes.flatMap(type => NAME_TO_TYPE_MAPPING[type] || [])]);
+      onEnvelope = envelope => {
+        for (const [header] of envelope[1]) {
+          if (header.type && types.has(header.type)) {
+            displayEnvelope(envelope);
```

Comment:
shouldn't we add a `break` here?

if we have an envelope with multiple items of the same type, that will display the same envelope multiple times

Reply:
Fixed, great catch!